dbms/src/Storages/KVStore/Region.h

```diff
@@ -235,15 +235,20 @@ class Region : public std::enable_shared_from_this<Region>
 
     Region() = delete;
 
+#ifdef DBMS_PUBLIC_GTEST
+public:
+#else
```

Comment:
what methods make us need to use this hack?

Reply:
`splitInto`, seems the friend class declared here never works.

Reply:
because `TEST_F(A, B)` will generate a subclass `class A_B` that inherits `class A` using macro and run the test body. So we need to workaround in another way. I've push a commit to remove the hack.